### PR TITLE
moved server creation from bin to lib; added example of client commun…

### PR DESCRIPTION
**Pull request by michaelmattig**
…icating with server

### CODE REVIEW

File: services/src/server.rs
Comment (michaelmattig) [inline services/src/server.rs:64]: can't use option.map_or_else here because I need the handler variable in both cases and with map_or_else it would be captured by the first closure

Comment (ChristianBeilschmidt) [reply services/src/server.rs:64]: Yeah, maybe this lint is too strict, we'll have to investigate it.

File: services/Cargo.toml
Comment (ChristianBeilschmidt) [inline services/Cargo.toml:25]: Yes, since you only use it in an example for now. We can move it back if we do Http requests from within with reqwest.

File: services/examples/request.rs
Comment (ChristianBeilschmidt) [inline services/examples/request.rs:5]:
```suggestion
/// Example of a client communicating with the geo engine
```

File: services/examples/request.rs
Comment (ChristianBeilschmidt) [inline services/examples/request.rs:14]: Isn't there a race condition that the server must start before executing `queries`?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
Just minor stuff to address before merging.

File: services/examples/request.rs
Comment (michaelmattig) [reply services/examples/request.rs:14]: it worked for me, but your are right. Would it be an acceptable solution to sleep the queries task for some time before issuing the first query or do we need a backchannel from the server that indicates that the server is running? Not sure if warp supports this though...

File: services/Cargo.toml
Comment (michaelmattig) [reply services/Cargo.toml]: 92699a3d13dbdd2eb312e95b07ae9ec5bc45c9dd

File: services/examples/request.rs
Comment (michaelmattig) [reply services/examples/request.rs]: 3304dc28317951e2ab2941d70b7815d22fdaca76

Comment (ChristianBeilschmidt) [reply services/examples/request.rs]: This works, it's just… couldn't you instead of using `join!`, which does things in parallel, first wait for a response of the server startup and then issue the queries?

Comment (michaelmattig) [reply services/examples/request.rs]: there is no response from the server. The server is spawned as a tokio task which is then awaited on. This await returns only when the server is shutdown. I haven't found a hook in warp that would allow being notified once the server is ready to serve requests and I think that's because usually a server isn't used from within the server executable it is run in.

Comment (ChristianBeilschmidt) [reply services/examples/request.rs]: Hm, but even MAPPING can say: I'm up and running, awaiting requests. Is there an example for that? We could postpone it as a different task though.

Comment (michaelmattig) [reply services/examples/request.rs]: that's what I'm checking in the wait method. if there is some response from the server.

Comment (ChristianBeilschmidt) [reply services/examples/request.rs]: This solution is like pressing F5 in a browser. I'm asking if there is any chance that our server can state internally that it runs.
I guess it can block until it receives incoming requests.
If not, then this is fine.

Comment (michaelmattig) [reply services/examples/request.rs]: I haven't found anything for querying the server state or even getting notified of successful startup. The documentation of `run` states
> Run this Server forever on the current thread.

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**
